src/scene/materials/material.js

```diff
@@ -239,12 +267,41 @@ class Material {
 
     // returns boolean depending on material being transparent
     get transparent() {
-        return this.blend;
+        return this._blendState.blend;
     }
 
+    // called when material changes transparency, for layer composition to add it to appropriate
+    // queue (opaque or transparent)
+    _markBlendDirty() {
+        if (this._scene) {
+            this._scene.layers._dirtyBlend = true;
+        } else {
+            this._dirtyBlend = true;
+        }
+    }
+
+    /**
+     * Controls how fragment shader outputs are blended when being written to the currently active
+     * render target. This overwrites blending type set using {@link pc.Material#blendType}, and
+     * offers more control over blending.
+     *
+     * @type { BlendState }
+     */
+    set blendState(value) {
+        if (this._blendState.blend !== value.blend) {
+            this._markBlendDirty();
+        }
+        this._blendState.copy(value);
```

Comment:
might as well move this into the if block above?

Reply:
I don't think I can, as that would only copy it blendState.blend is different, but it needs to be copied even if that is equal and other members of it are not equal.